knack/commands.py

```diff
@@ -124,6 +124,8 @@ def update_argument(self, param_name, argtype):
         # that coincides with the default
         if isinstance(arg_default, str):
             arg_default = DefaultStr(arg_default)
+        elif isinstance(arg_default, bool):
+            pass
```

Comment:
> I don't think we can simply `pass` here, as this will make it impossible to distinguish whether the received `True` is user specified or the default value.

It only pass the converting of bool to int. Not pass the default value

Reply:
I think it's useless, because it only added `is_default` property for str and int, not cover the other types, such as bool, float, timestamp etc.

Reply:
BTW, I also found these lines has a bug when the arg_default is `0` int value https://github.com/microsoft/knack/blob/e496c9590792572e680cb3ec959db175d9ba85dd/knack/commands.py#L129-L131

Reply:
Could you elaborate what the bug is?

Reply:
After reviewing the code, I feel this `if arg_default:` statement is deliberately added in order to ignore `False`/`0` as default.

For example, for arguments `--disable-integrity-monitoring` and `--disable-integrity-monitoring-autoupgrade` of `az vm create`, their default values are `False`/`0`, so they are deliberately omitted from the help message:

```
> az vm create -h
...
    --disable-integrity-monitoring                      : Disable the default behavior of installing
                                                          guest attestation extension and enabling
                                                          System Assigned Identity for Trusted
                                                          Launch enabled VMs and VMSS.
    --disable-integrity-monitoring-autoupgrade          : Disable auto upgrade of guest attestation
                                                          extension for Trusted Launch enabled VMs
                                                          and VMSS.
```